### ISSUE TITLE
Location List Issue - Vim 7.4, patches 1-1707

### ISSUE THREAD

**Issue by amadeus** [scrooloose/syntastic#1738]
I am using the latest version of Vim, built from source, and the location list appears broken when attempting to use it to jump to a location of an error/warning returned from ESLint.

When I press enter on a line in the location list, my cursor just returns to the last position it was in the buffer before I jumped to the location list split.

I also get an error:

```
E926: Current location list was changed
```

Not sure if this is an issue Vim needs to fix, or if they changed something with how location lists are handled and Syntastic needs a fix.

**Comment by n8henrie** [comment]
:+1: http://superuser.com/questions/1059194/vim-syntastic-not-jumping-to-the-error

Something definitely changed with vim; old versions are working fine. Also `lnext` and `lprev` still work, just not selecting from the list.

I haven't seen any [vim issues](https://github.com/vim/vim/issues) raised on it yet.

**Comment by lcd047** [comment]
Yes, this behaviour was introduced by Vim 7.4.1592, but it isn't clear the bug is in Vim, or that it should be addressed there.  The patch 1592 is a bug fix, and it solves a real crash.  On the other hand, what syntastic does on the relevant code path is a mess, and it's definitely wrong in some scenarios.  I made several attempts over the years to fix this, but the limitations in Vim's API basically mean that fixing some scenario breaks others.  Until patch 1592 the broken scenarios were just less common.  I'll try again, but I'm not particularly confident the result is going to be substantially better this time. _shrug_

**Comment by lcd047** [comment]
Duplicate of #1127.

**Comment by lcd047** [comment]
Please try 7e986f1 and report any problems.  Segfaults might be on the plate too (well, hopefully not).

**Comment by amadeus** [comment]
I am testing it out, so far it seems to work great, thanks!

**Comment by n8henrie** [comment]
Sorry for the late reply -- working flawlessly for me on vim 7.4.1707.

**Comment by luan** [comment]
@lcd047 I'm getting this error when opening a file for the first time through the locationlist:

```
Error detected while processing function <SNR>114_BufEnterHook[5]..342[20]..269[2]..270[5]..330[2]..329:               line    8:                                                                                                             E121: Undefined variable: b:syntastic_changedtick
```

**Comment by lcd047** [comment]
@luan Right, please upgrade to d9814c5.

**Comment by lcd047** [comment]
Not fixed, see the end of #1127 for updates.
